packages/subgraph/src/util.ts

```diff
@@ -212,6 +212,79 @@ export function buyOrSellETH(userAddr: string, amount: BigDecimal): void {
   position.save();
 }
 
+export function handleOSQTHChange(userAddr: string, amount: BigDecimal): void {
+  if (amount.equals(ZERO_BD) || userAddr == null) return;
+
+  let position = loadOrCreatePosition(userAddr);
+  let osqthPrices = getoSQTHETHPrice();
+
+  // When position side chages, reset PnLs and calculate with remaining amount
+  let newAmount = position.currentOSQTHAmount.plus(amount);
+  if (position.currentOSQTHAmount.times(newAmount).lt(ZERO_BD)) {
+    position = initPosition(userAddr, position);
+    amount = newAmount;
+  }
+
+  let absAmount = amount.lt(ZERO_BD) ? amount.neg() : amount;
+  let isLong =
```

Comment:
please use the formula of `(position.currentOSQTHAmount.gt(account.accShortAmount.toBigDecimal()` instead of just using osqthAmount to check the position type

Reply:
long means oSQTH in wallet - oSQTH debt in vault > 0, short means `oSQTH in wallet - oSQTH debt in vault < 0`

Reply:
Oh, totally agree.